In [39]:
import pandas as pd
import numpy as np
from sklearn.linear_model import Perceptron
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import accuracy_score

In [11]:
data = pd.read_csv('perceptron-train.csv', names = ['1','2','3'])

In [12]:
test = pd.read_csv('perceptron-test.csv', names = ['1','2','3'])

In [15]:
scaler = StandardScaler()

In [24]:
#разобьем обущающую выборку на X_train и y_train
X_train = data[['2','3']]
y_train = data['1']

In [25]:
#разобьем тестовую выборку на X_test и y_test
X_test = test[['2','3']]
y_test = test['1']

In [33]:
#Обучим Персептрон
model = Perceptron(random_state=241)

In [35]:
model.fit(X_train, y_train)

Perceptron(alpha=0.0001, class_weight=None, early_stopping=False, eta0=1.0,
      fit_intercept=True, max_iter=None, n_iter=None, n_iter_no_change=5,
      n_jobs=None, penalty=None, random_state=241, shuffle=True, tol=None,
      validation_fraction=0.1, verbose=0, warm_start=False)

In [38]:
#спрогнозируем модель
predictions = model.predict(X_test)

In [49]:
as1 = accuracy_score(y_test, predictions)
as1

0.655

In [45]:
#нормализуем признаки
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)

In [46]:
# фит - предикт на стандартизованных признаков
model.fit(X_train_scaled, y_train)
predictions_scaled = model.predict(X_test_scaled)

In [50]:
as2 = accuracy_score(y_test, predictions_scaled)
as2

0.845

In [55]:
#посмотрим как улучшилась модель предсказания
Res_normal = as2-as1
Res_normal

0.18999999999999995

# Нормализация данных дает значительный прирост в качестве — оно увеличивается на 19%.